src/debugger/ios/iOSPlatform.ts

```diff
@@ -35,29 +35,28 @@ export class IOSPlatform implements IAppPlatform {
                 runArguments.push(this.simulatorTarget);
             }
 
-            return new CommandExecutor(this.projectPath).spawnReactCommand("run-ios", runArguments).then((runIos) => {
-                const deferred = Q.defer<void>();
-                runIos.on("error", (err: Error) => {
+            const runIos = new CommandExecutor(this.projectPath).spawnReactCommand("run-ios", runArguments);
+            const deferred = Q.defer<void>();
+            runIos.on("error", (err: Error) => {
+                deferred.reject(err);
+            });
+            runIos.stderr.on("data", (data: Buffer) => {
+                const dataString = data.toString();
+                if (dataString.indexOf("No devices are booted") !== -1 // No emulators are started
+                    || dataString.indexOf("FBSOpenApplicationErrorDomain") !== -1) { // The incorrect emulator is started
+                    deferred.reject(new Error("Unable to launch iOS simulator. Try specifying a different target."));
+                }
+            });
+            runIos.on("exit", (code: number) => {
+                if (code !== 0) {
+                    const err = new Error(`Command failed with exit code ${code}`);
+                    Log.commandFailed(["react-native", "run-ios"].concat(runArguments).join(" "), err);

```

Comment:
You are right. I won't fix this in this PR though, I'm going to fix it in my next PR where I'm solving the errors printing twice issue: https://github.com/Microsoft/vscode-react-native/issues/55